packages/dashboard/src/pages/Wallets/components/AddTokenDialog/index.tsx

```diff
@@ -0,0 +1,309 @@
+                        onNext={() => setStep(AddTokenStep.CONFIRM)}
+                    />
+                ) : null}
+                {step === AddTokenStep.CONFIRM ? (
+                    <>
+                        <AddTokenConfirmUI
+                            token={token}
+                            balance={balance}
+                            onBack={() => setStep(AddTokenStep.INFORMATION)}
+                            onConfirm={onSubmit}
+                        />
+                    </>
+                ) : null}
+            </MaskDialog>
+        )
+    },
+)
+
+export interface AddTokenFormUIProps {
+    open: boolean
+    address: string
+    exclude: string[]
+    onAddressChange: (address: string) => void
+    onNext: () => void
+    onClose: () => void
+    token?: ERC20TokenDetailed
+}
+
+export const AddTokenFormUI = memo<AddTokenFormUIProps>(
+    ({ address, onAddressChange, token, open, exclude, onClose, onNext }) => {
+        const t = useDashboardI18N()
+        const classes = useStyles()
+        const [symbol, setSymbol] = useState('')
+        const [decimals, setDecimals] = useState('')
+
+        const validateAddressMessage = useMemo(() => {
+            if (address.length && !EthereumAddress.isValid(address)) return t.wallets_incorrect_address()
+            if (exclude.find((item) => item === address)) return t.wallets_token_been_added()
+            return ''
+        }, [address])
+
+        const validateSymbolMessage = useMemo(() => {
+            if (symbol.length && symbol.length > 11) return t.wallets_token_symbol_tips()
+            return ''
+        }, [])
+
+        const validateDecimalsMessage = useMemo(() => {
+            if (decimals.length && (Number(decimals) < 0 || Number(decimals) > 18))
+                return t.wallets_token_decimals_tips()
+            return ''
+        }, [decimals])
+
+        const disableButton = useMemo(
+            () =>
+                !address ||
+                !symbol ||
+                !decimals ||
+                !!validateAddressMessage ||
+                !!validateSymbolMessage ||
+                !!validateDecimalsMessage,
+            [address, symbol, decimals, validateAddressMessage, validateSymbolMessage, validateDecimalsMessage],
+        )
+
+        useEffect(() => {
+            if (token) {
+                setSymbol(token.symbol ?? '')
+                setDecimals(String(token.decimals))
+            }
+        }, [token])
+
+        //#region when dialog be closed, clear input
+        useUpdateEffect(() => {
+            if (!open) {
+                setSymbol('')
+                setDecimals('')
+            }
+        }, [open])
+
+        return (
+            <>
+                <DialogContent className={classes.content}>
+                    <form>
+                        <div className={classes.item}>
+                            <label className={classes.title}>{t.wallets_add_token_contract_address()}</label>
```

Comment:
Please wrap it with `Typography` otherwise the font doesn't look right. Use `label` on the TextField if possible.